tools/extra/fpgadiag/eth_group.cpp

```diff
@@ -54,140 +54,49 @@
-		printf("Doen't support VFIO_TYPE1_IOMMU \n");
-		return -1;
+	res = opae_uio_region_get(&uio, 0, (uint8_t **)&mem, NULL);
+	if (res) {
+		return res;
 	}
 
-	if (snprintf(dev_path, sizeof(dev_path),
-		"%s/%d", "/dev/vfio", vfio_id) < 0) {
-		printf("snprint fails \n");
-		return -1;
-	}
-
-	group = open(dev_path, O_RDWR);
-	if (group < 0) {
-		fprintf(stderr, "error opening group: %s\n",
-			strerror(errno));
-		return -1;
-	}
-
-	// checks VFIO group status
-	ioctl(group, VFIO_GROUP_GET_STATUS, &group_status);
-	if (!(group_status.flags & VFIO_GROUP_FLAGS_VIABLE)) {
-		printf("Wrong VFIO_GROUP_FLAGS_VIABLE \n");
-		return -1;
-	}
 
-	// Add the group to the container
-	ioctl(group, VFIO_GROUP_SET_CONTAINER, &container);
+	mmap_ptr = mem;
+	ptr_ = (uint64_t*)mem;
 
-	// Enable the IOMMU model
-	ioctl(container, VFIO_SET_IOMMU, VFIO_TYPE1_IOMMU);
+	eth_info.csr = *(ptr_ + 1);
+	eth_dfh.csr = *(ptr_);
+	direction = eth_info.direction;
+	phy_num = eth_info.no_phys;
+	group_id = eth_info.group_num;
+	speed = eth_info.speed_gbs;
+	df_id = eth_dfh.id;
+	eth_lwmac = eth_info.light_wight_mac;
 
-	// Get IOMMU info
-	ioctl(container, VFIO_IOMMU_GET_INFO, &iommu_info);
-
-	// Allocate some space and setup a DMA mapping 
-	dma_map.vaddr = (unsigned long int) mmap(0, 1024 * 1024, PROT_READ | PROT_WRITE,
-		MAP_PRIVATE | MAP_ANONYMOUS, 0, 0);
-	dma_map.size = 1024 * 1024;
-	dma_map.iova = 0;
-	dma_map.flags = VFIO_DMA_MAP_FLAG_READ | VFIO_DMA_MAP_FLAG_WRITE;
-
-	ioctl(container, VFIO_IOMMU_MAP_DMA, &dma_map);
-
-	// Get FIO a file descriptor for the device
-	device = ioctl(group, VFIO_GROUP_GET_DEVICE_FD, fpga_mdev_str.c_str()	);
-	if (device < 0) {
-		fprintf(stderr, "error getting device fd: %s\n",
-			strerror(errno));
+	// Reset Mac
+	if (!mac_reset()) {
+		printf("Failed to reset MAC \n");
 		return -1;
 	}
-	// Get dvice info
-	ioctl(device, VFIO_DEVICE_GET_INFO, &device_info);
-
-	for (i = 0; i < (int)device_info.num_regions; i++) {
-		struct vfio_region_info reg;
-		memset(&reg, 0, sizeof(reg));
-		reg.argsz = sizeof(reg);
-		reg.index = i;
-		ioctl(device, VFIO_DEVICE_GET_REGION_INFO, &reg);
-		uint8_t *mem;
-		mem = (uint8_t*) mmap(NULL, reg.size, PROT_READ | PROT_WRITE,
-			MAP_SHARED, device, reg.offset);
-
-		if (mem == MAP_FAILED) {
-			printf("Failed to MMAP \n");
-			eth_group_close();
-			return -1;
-		}
 
-		reg_size = reg.size;
-		reg_offset = reg.offset;
-
-		mmap_ptr = mem;
-		ptr_ = (uint64_t*)mem;
-
-		eth_info.csr = *(ptr_ + 1);
-		eth_dfh.csr = *(ptr_);
-		direction = eth_info.direction;
-		phy_num = eth_info.no_phys;
-		group_id = eth_info.group_num;
-		speed = eth_info.speed_gbs;
-		df_id = eth_dfh.id;
-		eth_lwmac = eth_info.light_wight_mac;
-
-		// Reset Mac
-		if (!mac_reset()) {
-			printf("Failed to reset MAC \n");
-			return -1;
-		}
-	}
 	return 0;
 }
 
 // reset mac
 bool eth_group::mac_reset()
 {
-	uint32_t i;
-	uint32_t value;
-	int ret = 0;
+	uint32_t i       = 0;
+	uint32_t value   = 0;
+	int ret           = 0;
```

Comment:
I don't understand why we need mac_value. Can this be simplified to something like this?

```c++
// reset mac
bool eth_group::mac_reset()
{
  for (uint32_t i = 0; i < phy_num; i++) {
    if (read_reg(ETH_GROUP_MAC, i, 0, MAC_CONFIG) == ETH_GROUP_RET_VALUE))
        return false;
    if (write_reg(ETH_GROUP_MAC, i, 0, MAC_CONFIG, 0x0))
      return false;
  return true;
}
```

Reply:
Removed  mac_value variable and simplified code